### PR TITLE
Additional mulligan support #5600

### PR DESCRIPTION
**Pull request by hitch17**
Adds the ability to select the mulligan type (Paris, Vancouver, London and Canadian Highlander) for your game.  Currently only set up in NewTableDialog.  Will add to NewTournamentDialog if things go well.  I have not tested this with [Serum Powder](https://magic.wizards.com/en/articles/archive/competitive-gaming/mythic-championship-ii-format-and-london-test-2019-02-21) yet.

### CODE REVIEW

**Comment by hitch17** [comment]
Related to https://github.com/magefree/mage/issues/5600

File: Mage/src/main/java/mage/game/mulligan/MulliganType.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    public Mulligan getMulligan(int freeMulligans) {
+        switch(this) {
+            case PARIS:
+                return new ParisMulligan(freeMulligans);
+            case CANADIAN_HIGHLANDER:
+                return new CanadianHighlanderMulligan(freeMulligans);
+            case LONDON:
+                return new LondonMulligan(freeMulligans);
+            default:
+            case VANCOUVER:
```
Comment (ingmargoudt) [inline Mage/src/main/java/mage/game/mulligan/MulliganType.java]: Can there be a case inside a default?

Comment (hitch17) [reply Mage/src/main/java/mage/game/mulligan/MulliganType.java]: The `default` case will fall through to the next option, which is VANCOUVER.  I could add an explicit return of VancouverMulligan inside the default option if you'd like.  Is there a specific concern to this pattern?

**Comment by hitch17** [comment]
Serum Powder does not work correctly with this version of London Mulligan.  In that wizards post, they suggest that the mulligan choice occurs after you've chosen cards to put on the bottom of your library.  If you've mulliganed twice, this means that you only get to use its ability after discarding down to 5 and getting 5 new cards (currently, this allows you to use its ability with the 7 cards in hand before discard).

File: Mage/src/main/java/mage/game/mulligan/MulliganType.java (hunk 1 above)
Comment (ingmargoudt) [reply Mage/src/main/java/mage/game/mulligan/MulliganType.java]: It is not a concern at all, it is just that I have never seen a dedicated case in a default . I even did not know it was possible. I do believe you it works though 👍 I always write:

case A:
 doThis();
case B:
  doThat();
default:
  doSomethingElse();

File: Mage.Tests/src/test/java/org/mage/test/multiplayer/BlatantThieveryTest.java [hunk 2]
```diff
@@ -24,7 +25,7 @@
 
     @Override
     protected Game createNewGameAndPlayers() throws GameException, FileNotFoundException {
-        Game game = new FreeForAll(MultiplayerAttackOption.MULTIPLE, RangeOfInfluence.ALL, 0, 20);
+        Game game = new FreeForAll(MultiplayerAttackOption.MULTIPLE, RangeOfInfluence.ALL, new VancouverMulligan(0), 20);
```
Comment (JayDi85) [inline Mage.Tests/src/test/java/org/mage/test/multiplayer/BlatantThieveryTest.java]: Default mulligan constructor can be replaced with standard call like `MulliganType.GAME_DEFAULT.getMulligan(0)` or same call (if wizards try to change default mulligan then no needs to change it all around).

**Comment by JayDi85** [comment]
@hitch17 don't add it to tournaments... There are compatibility issue in table/tournament netbeans editor (it's not your PR, it's from old changes -- some controls was added without editor and destroys on save).

I'll fix that problem and add mulligan field to tournament form too.

**Comment by hitch17** [comment]
> @hitch17 don't add it to tournaments... There are compatibility issue in table/tournament netbeans editor (it's not your PR, it's from old changes -- some controls was added without editor and destroys on save).
> 
> I'll fix that problem and add mulligan field to tournament form too.

Sounds good.  Thanks for the heads up.

**Pull request merged**